samza-sql/src/main/java/org/apache/samza/sql/translator/ProjectTranslator.java

```diff
@@ -53,28 +60,42 @@
     this.queryId = queryId;
   }
 
-  private static class ProjectMapFunction implements MapFunction<SamzaSqlRelMessage, SamzaSqlRelMessage> {
+  @VisibleForTesting
+  public static class ProjectMapFunction implements MapFunction<SamzaSqlRelMessage, SamzaSqlRelMessage> {
     private transient Project project;
     private transient Expression expr;
     private transient TranslatorContext context;
+    private transient MetricsRegistry metricsRegistry;
+    private transient SamzaHistogram processingTimeHistogram; // milli-seconds
```

Comment:
nit: As a general practice, Samza code base doesn't have type related details to the variable Name. You can just call it as processingTime and numMessages or messages.

Reply:
done